typings/index.d.ts

```diff
@@ -2230,6 +2232,10 @@ declare module 'discord.js' {
 
   interface ChannelData {
     name?: string;
+    type?: Exclude<
+      keyof typeof ChannelType | ChannelType,
+      'dm' | 'group' | 'unknown' | ChannelType.dm | ChannelType.group | ChannelType.unknown
+    >;
```

Comment:
the method only accepts the string varient, perhaps maybe `'news' | 'text'` (since they are the only ones that can be changed to at the moment)

Reply:
I was originally not in favour of restricting the types (responsibility of the user), but I guess this exclusion already partially does that anyway. Will look into how best to do that